### DISCUSSION TITLE
What is right choice for deployment of Semantic Kernel using python?

### DISCUSSION THREAD

**Discussion by lovedeepatsgit** [microsoft/semantic-kernel#10034] [Q&A] [answered]
I have developed semantic kernel code, which uses semantic kernels agentic framework using python. What I am facing issues and need some ideas b which I can use that code for production? So If anyone have some particular suggestion or recommendations, the please reply. And, if there is any proper documentation to do this?

**Comment by moonbox3** [comment, marked as answer]
Hi @lovedeepatsgit, there is no "one-size fits all" answer when it comes to deployments -- we have customers deploying SK as part of a web app, as a docker image and either a FastAPI or a Flask app, as an Azure Function or a micro service (usually on the .Net side), among others. Semantic Kernel is designed to be the middleware, so you're free to deploy it any way you see fit.

**Reply by lovedeepatsgit** [reply]
Okay great to hear from your side @moonbox3.
